ktor-client/ktor-client-android/jvm/src/io/ktor/client/engine/android/AndroidClientEngine.kt

```diff
@@ -95,7 +95,7 @@ public class AndroidClientEngine(override val config: AndroidEngineConfig) : Htt
 
             val responseBody: Any = if (needToProcessSSE(data, statusCode, responseHeaders)) {
                 DefaultClientSSESession(
-                    data.body as SSEClientContent,
+                    outgoingContent as SSEClientContent,
                     content,
                     callContext
                 )
```

Comment:
This `needsToProcessSSE` could also be abstracted to an interface like...

```kotlin
interface ResponseAdapter {
    fun adapt(status: HttpStatusCode, headers: Headers, body: Any): Any?
}
```

Then do something like `responseBody: Any = data.attributes.getOrNull(ResponseAdapter)?.adapt(statusCode, headers, body) ?: content`.